tests/run/i3812.scala

```diff
@@ -0,0 +1,15 @@
+object Test {
+  def main(args: Array[String]): Unit = {
+    val x = 42
+    val Y = "42"
+
+    x match { case { 42 }           => () } // ok
+    x match { case { "42".toInt }   => () } // ok
+    x match { case { "42" }.toInt   => () } // ok
+    x match { case { "42".toInt }   => () } // ok
+    x match { case { Y.toInt }      => () } // ok
+    x match { case { Y }.toInt      => () } // ok
+
+    // x match { case Y.toInt          => () } // compiles but rejected by Ycheck
```

Comment:
@liufengyun `Ycheck` fails on this one. Should this be allowed?

Reply:
Good catch, I think it should be allowed.

Reply:
Note that `case "42".toInt =>` does not parse. It might be confusing that one is allowed but not the other

Reply:
The syntax specification requires the prefix has to be a` path` or `block`.

```
SimplePattern     ::=  PatVar                                                   Ident(wildcard)
                    |  Literal                                                  Bind(name, Ident(wildcard))
                    |  ‘(’ [Patterns] ‘)’                                       Parens(pats) Tuple(pats)
                    |  XmlPattern
                    |  SimplePattern1 [TypeArgs] [ArgumentPatterns]
SimplePattern1    ::=  Path
                    |  ‘{’ Block ‘}’
                    |  SimplePattern1 ‘.’ id
```